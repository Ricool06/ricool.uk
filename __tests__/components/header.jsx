import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from 'react-testing-library';
import Header from '../../src/components/header';

jest.mock('../../src/components/social-links.jsx', () => 'div');

describe('Header', () => {
  afterEach(cleanup);

  it('should have correct header text, set by title', () => {
    const { container } = render(<Header siteTitle={'Ricool\'s Blog'} />);

    expect(container.textContent).toBe('Ricool\'s Blog');
  });

  it('should match snapshot', () => {
    const jsonSnapshot = renderer.create(<Header siteTitle={'Ricool\'s Blog'} />).toJSON();
    expect(jsonSnapshot).toMatchSnapshot();
  });
});
