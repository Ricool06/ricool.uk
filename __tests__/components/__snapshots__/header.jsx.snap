// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div
  style={
    Object {
      "background": "#339988",
      "marginBottom": "1.45rem",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "1.45rem 1.0875rem",
      }
    }
  >
    <h1
      style={
        Object {
          "flex": "1 1 auto",
          "margin": 0,
        }
      }
    >
      <a
        href="/"
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
          }
        }
      >
        Ricool's Blog
      </a>
    </h1>
    <div />
  </div>
</div>
`;
